### ISSUE TITLE
pytest-cov warnings when used with parallel pytest-pylint runs

### ISSUE THREAD

**Issue by GergelyKalmar** [pytest-dev/pytest-cov#424]
# Summary
When running pytest-cov and pytest-pylint together I see errors like these:

```
Coverage.py warning: Data file '/home/gregory/Projects/coverage-test/.coverage.flux.42281.732185' doesn't seem to be a coverage data file: Couldn't use data file '/home/gregory/Projects/coverage-test/.coverage.flux.42281.732185': no such table: coverage_schema
Coverage.py warning: Data file '/home/gregory/Projects/coverage-test/.coverage.flux.42280.532559' doesn't seem to be a coverage data file: Couldn't use data file '/home/gregory/Projects/coverage-test/.coverage.flux.42280.532559': no such table: coverage_schema
```

## Expected vs actual result
I would expect that there are no such errors.

# Reproducer

## Versions
I've been running things with Python 3.8 under Ubuntu 20.04 LTS.

``requirements.txt``
```
coverage[toml]==5.2.1
pylint==2.5.3
pytest==6.0.1
pytest-cov==2.10.1
pytest-pylint==0.17.0
```

## Config
``pyproject.toml``
```
[tool.pytest.ini_options]
addopts = '--pylint --cov'

[tool.pylint.MASTER]
jobs = 8

[tool.coverage.run]
source = ['.']
```

## Code
``lorem.py:``
```
def ipsum():
    pass
```

Create a virtual env, install the packages from ``requirements.txt`` and run ``pytest``. On the first run the errors shall appear. Afterwards they can be triggered again by clearing the cache (``rm -r .coverage* .pytest_cache``).

The error doesn't appear with ``jobs = 1`` or when pylint isn't run.

**Comment by GergelyKalmar** [comment]
@ionelmc finally got the time to create a minimal example. May or may not be related to #250.

**Comment by ionelmc** [comment]
Nothing I can do about it in pytest-cov, pylint doesn't have proper multiprocessing pool shutdown. See https://pytest-cov.readthedocs.io/en/latest/subprocess-support.html#if-you-use-multiprocessing-pool and https://github.com/PyCQA/pylint/blob/707fc4603b9d505bda68cf6adf9805882b99d73c/pylint/lint/check_parallel.py#L82

You might want to make your report here https://github.com/PyCQA/pylint/issues instead.

**Comment by GergelyKalmar** [comment]
Thank you for the quick reply! It seems there are some other issues with pylint's multiprocessing (see e.g. https://github.com/PyCQA/pylint/issues/3584), so perhaps I'll just turn off pylint multiprocessing completely for now. Or I will go back and split up my test suite to run pylint, pytest and mypy separately through a Makefile (oh the horror) 1995 style, as it seems I can't possibly get all of these to work nicely together in a pytest session anyways. What a holy mess we have! :)

**Comment by The-Compiler** [comment]
> Or I will go back and split up my test suite to run pylint, pytest and mypy separately through a Makefile (oh the horror) 1995 style, as it seems I can't possibly get all of these to work nicely together in a pytest session anyways. What a holy mess we have! :)

FWIW [tox](https://tox.readthedocs.io/en/latest/) is the quasi-standard solution for this.

**Comment by GergelyKalmar** [comment]
That's fair. I figured though that I wouldn't be able to focus the test runs that nicely (as I do with ``pytest -k thing_I_work_on`` during development right now) if I start taking things out of the pytest session, so in the end I went with the no-multiprocessing option and tried to work around the issues with pytest plugins generally. Some are rather annoying (like this one related to pylint: https://github.com/PyCQA/astroid/issues/703), but I guess that's the best one can do at the moment.

It is a bit weird that pytest-pylint would interfere with pytest-cov though, I thought pytest-pylint does its job independently before other things are executed.
